runtime/vm/callin.cpp

```diff
@@ -942,6 +942,54 @@ sendForGenericInvoke(J9VMThread *currentThread, j9object_t methodHandle, j9objec
 	Trc_VM_sendForGenericInvoke_Exit(currentThread);
 }
 
+void JNICALL
+sendResolveMethodHandle2(J9VMThread *currentThread, J9ConstantPool *ramCP, UDATA cpIndex, I_32 refKind, J9Class *resolvedClass, J9ROMNameAndSignature* nameAndSig, j9object_t appendix)
+{
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	J9VMEntryLocalStorage newELS;
+	if (buildCallInStackFrame(currentThread, &newELS, true, false)) {
+		/* Convert name and signature to String objects */
+		J9JavaVM *vm = currentThread->javaVM;
+		J9MemoryManagerFunctions const * const mmFuncs = vm->memoryManagerFunctions;
+		J9UTF8 *nameUTF = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+		j9object_t nameString = mmFuncs->j9gc_createJavaLangString(currentThread, J9UTF8_DATA(nameUTF), J9UTF8_LENGTH(nameUTF), J9_STR_INTERN);
```

Comment:
What preserves the `appendix` object pointer across this call?  If a GC runs, how do you prevent it from being corrupted?

Likely need to add something like:
```
PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, appendix);
```

Reply:
fixed